daemon/health.go

```diff
@@ -188,15 +188,27 @@ func handleProbeResult(d *Daemon, c *container.Container, result *types.Healthch
 func monitor(d *Daemon, c *container.Container, stop chan struct{}, probe probe) {
 	probeTimeout := timeoutWithDefault(c.Config.Healthcheck.Timeout, defaultProbeTimeout)
 	probeInterval := timeoutWithDefault(c.Config.Healthcheck.Interval, defaultProbeInterval)
+loop:
 	for {
 		select {
 		case <-stop:
 			logrus.Debugf("Stop healthcheck monitoring for container %s (received while idle)", c.ID)
 			return
 		case <-time.After(probeInterval):
-			logrus.Debugf("Running health check for container %s ...", c.ID)
 			startTime := time.Now()
+			healthChecksWaiters.Inc()
 			ctx, cancelProbe := context.WithTimeout(context.Background(), probeTimeout)
+			if err := d.healthcheckLimiter.Acquire(ctx, 1); err != nil {
+				healthChecksWaiters.Dec()
+				// if we see this, we have ourselves a nasty bug
```

Comment:
Maybe a counter for failures here?

Reply:
Doesn't necessarily need to be added in this PR unless you push another update.

Reply:
Added